pkg/resources/tag_association_acceptance_test.go

```diff
@@ -32,29 +41,108 @@ func TestAcc_TagAssociation(t *testing.T) {
 		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
 			tfversion.RequireAbove(tfversion.Version1_5_0),
 		},
-		CheckDestroy: nil,
+		CheckDestroy: acc.CheckTagValueEmpty(t),
 		Steps: []resource.TestStep{
 			{
 				ConfigDirectory: acc.ConfigurationDirectory("TestAcc_TagAssociation/basic"),
-				ConfigVariables: m(),
-				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr(resourceName, "object_type", "DATABASE"),
+				ConfigVariables: m(tagId, tagValue),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "id", helpers.EncodeSnowflakeID(tagId.FullyQualifiedName(), tagValue, string(sdk.ObjectTypeDatabase))),
+					resource.TestCheckResourceAttr(resourceName, "object_type", string(sdk.ObjectTypeDatabase)),
+					resource.TestCheckResourceAttr(resourceName, "object_identifiers.#", "1"),
+					resource.TestCheckTypeSetElemAttr(resourceName, "object_identifiers.*", acc.TestClient().Ids.DatabaseId().FullyQualifiedName()),
+					resource.TestCheckResourceAttr(resourceName, "tag_id", tagId.FullyQualifiedName()),
+					resource.TestCheckResourceAttr(resourceName, "tag_value", tagValue),
+				),
+			},
+			// external change
```

Comment:
external change can be either unset or setting to different value which are a bit different, please add also this one.

Reply:
Added.